cdap-docs/tools/doc-cdap-default.py

```diff
@@ -0,0 +1,434 @@
+        keys.remove(key)
+    keys.sort()        
+    keys = FIRST_TWO_SECTIONS + keys
+    for key in keys:
+        xml += XML_SECTION_SUB % key
+        props = defaults[key]
+        props.sort(key = lambda p: p.name)
+        for prop in props:
+            xml += XML_PROP_OPEN
+            xml += XML_NAME_SUB % prop.name
+            xml += XML_VALUE_SUB % prop.value
+            xml += XML_DESCRIP_OPEN
+            if prop.description:
+                for line in textwrap.wrap(prop.description):
+                    xml += XML_DESCRIP_SUB % line
+            else:
+                print "WARNING: No description for property %s" % prop.name
+                if prop.name in exclusions:
+                    print "but in the list of exclusions"
+            xml += XML_DESCRIP_CLOSE
+            if prop.final:
+                xml += XML_FINAL
+            xml += XML_PROP_CLOSE
+    xml += XML_CONFIG_CLOSE
+    
+    filepath = default_xml_filepath('_revised.xml')
+    f = open(filepath, 'wb')
+    f.write(XML_HEADER.encode('utf8'))
+    f.write(xml)
+    f.close()
+
+def load_create_rst(filesource='', filepath=''):
+    defaults = load_defaults(filesource)
+    table = create_rst(defaults)
+    if filepath:
+        f = open(filepath, 'w')
+        f.write(table)
+        f.close()
+    else:
+        print table
+        
+def load_xml(props_only=False):
+    print 'Loading cdap-default.xml'
+    items = load_defaults('')
+    section_counter = 0
+    properties_counter = 0
+    for item in items:
+        if str(item.__class__) == '__main__.Section':
+            section_counter += 1
+            if not props_only:
+                item.display()                
+        else:
+            properties_counter += 1
+            item.display()
+    print "Sections: %d\nProperties: %d" % (section_counter, properties_counter)
+
+def print_usage():
+    print """Usage:
+    --rst [source] [target]  Loads the existing default XML and creates an RST file from it
+                             source: the XML to be loaded, if not the default
+                             [optional, but required if target specified]
+                             target: where to write the rst, if not standard output [optional]
+
+      If both options are not specified, reads from default and writes to standard output                      
+
+    --rebuild  Loads the existing default XML and rewrites it in the correct order to
+               a file at the same location with '_revised.xml'
+
+    --load-props  Loads the existing default XML and writes the properties to standard output
+    --load-xml    Loads the existing default XML and writes it to standard output
+    --print-rst   Loads the existing default XML and writes the RST to standard output
+
+    --help  This help statement
+"""
+
+def main():
+    """ Main program entry point.
+    """
+    if len(sys.argv) == 1:
+        print_usage()
+        return
+    elif sys.argv[1] == '--rebuild':
```

Comment:
can we use optparse or something similar to parse arguments, instead of doing it ourselves? 

One of my concerns is there are no named single character alternatives for each of these arguments.